connections.md

```diff
@@ -1,8 +1,252 @@
+* The `label` attribute provides a suggested label for the DID being exchanged. This allows the user to tell multiple exchange requests apart. This is not a trusted attribute.
+* The `connection` attribute contains the `did` and `did_doc` attributes. This format maintains consistency with the Response message where this attribute is signed.
+* The `did` indicates the DID being exchanged.
+* The `did_doc~attach`  attachment is required when `did` contains a peer did. contains the DID Doc associated with the DID. If the DID method for the presented DID is not a peer method and the DID Doc is resolvable on a ledger, the `did_doc` attribute is optional. The attachment must include a signature
+
+##### Correlating requests to invitations
+
+An invitation is presented in one of two forms:
+
+* An explicit OutOfBand invitation with its own `@id`.
+* An [implicit](#implicit-invitation) invitation contained in a DID document's [`service`](https://w3c-ccg.github.io/did-spec/#service-endpoints) attribute.
+
+When a `request` responds to an explicit invitation, its `~thread.pthid` MUST be equal to the `@id` property of the invitation.
+
+When a `request` responds to an implicit invitation, its `~thread.pthid` MUST contain a [DID URL](https://w3c-ccg.github.io/did-spec/#dfn-did-url) that resolves to the specific `service` on a DID document that contains the invitation.
+
+TODO: Include examples using a ledger resolvable DID.
+
+**Example referencing an explicit invitation**
+
+```json
+{
+  "@id": "a46cdd0f-a2ca-4d12-afbf-2e78a6f1f3ef",
+  "@type": "https://didcomm.org/didexchange/1.0/request",
+  "~thread": { "pthid": "032fbd19-f6fd-48c5-9197-ba9a47040470" },
+  "label": "Bob",
+  "did": "B.did@B:A",
+  "did_doc~attach": {
+    "mime-type": "",
+    "data": {
+      "base64": "<b64 encoded bytes of diddoc>"// DID Doc contents here.
+    }
+    "sig": <sig info>
+  }
+}
+```
+
+**Example referencing an implicit invitation**
+
+```json
+{
+  "@id": "a46cdd0f-a2ca-4d12-afbf-2e78a6f1f3ef",
+  "@type": "https://didcomm.org/didexchange/1.0/request",
+  "~thread": { "pthid": "did:example:21tDAKCERh95uGgKbJNHYp#invitation" },
+  "label": "Bob",
+  "did": "B.did@B:A",
+  "did_doc~attach": {
+    "mime-type": "",
+    "data": {
+      "base64": "<b64 encoded bytes of diddoc>"// DID Doc contents here.
+    }
+    "sig": <sig info>
+  }
+}
+```
+
+
+##### Request Transmission
+
+The Request message is encoded according to the standards of the Encryption Envelope, using the `recipientKeys` present in the invitation.
+
+If the `routingKeys` attribute was present and non-empty in the invitation, each key must be used to wrap the message in a forward request, then encoded in an Encryption Envelope. This processing is in order of the keys in the list, with the last key in the list being the one for which the `serviceEndpoint` possesses the private key.
+
+The message is then transmitted to the `serviceEndpoint`.
```

Comment:
Good to add an example here, something like `https://cloud-agent.example.com/...`